core/src/org/pentaho/di/core/variables/Variables.java

```diff
@@ -40,7 +40,7 @@
  * @author Sven Boden
  */
 public class Variables implements VariableSpace {
-  private final ConcurrentHashMap<String, String> properties = new ConcurrentHashMap<String, String>();
+  private final Map<String, String> properties = new ConcurrentHashMap<>();
```

Comment:
Shouldn't this be:
 private final Map<String, String> properties = new ConcurrentHashMap<>( <String>, <String> );